### PR TITLE
BUG: make np.transpose return a view of the mask

### PR DESCRIPTION
**Pull request by eric-wieser**
Previously a copy of the data, but view of the mask.

<s>Deal with #9432 first, as this builds on that</s>

### CODE REVIEW

**Comment by charris** [comment]
Needs rebase.

**Comment by eric-wieser** [comment]
Not until #9432 is merged

File: numpy/ma/core.py [hunk 1]
```diff
@@ -5748,14 +5741,15 @@ def take(self, indices, axis=None, out=None, mode='raise'):
...
-    T = property(fget=lambda self: self.transpose())
-    swapaxes = _arraymethod('swapaxes')
     clip = _arraymethod('clip', onmask=False)
     copy = _arraymethod('copy')
+    diagonal = _arraymethod('diagonal')
+    flatten = _arraymethod('flatten')
+    repeat = _arraymethod('repeat')
     squeeze = _arraymethod('squeeze')
+    swapaxes = _arraymethod('swapaxes')
+    T = property(fget=lambda self: self.transpose())
```
Comment (mhvk) [inline numpy/ma/core.py]: I know this was there before, but what is wrong with `fget=transpose` (after the definition just below)?

Comment (eric-wieser) [inline numpy/ma/core.py]: Subclass behaviour?

Comment (mhvk) [inline numpy/ma/core.py]: Ah, indeed. Yes, better leave as is...

**Comment by eric-wieser** [comment]
Ok, finally got around to rebasing @charris

**Comment by ahaldane** [comment]
LGTM. I think I will go ahead and merge later today.

**Comment by ahaldane** [comment]
Merging. Thanks Eric!

**Pull request merged**